### ISSUE TITLE
[FE] Add aria labels to new appointment list component

### ISSUE THREAD

**Issue by wullaski** [department-of-veterans-affairs/va.gov-team#66707]
The unified appointment list should have appropriate aria labels to improve accessibility.

See sketch file annotations:
https://www.sketch.com/s/0e890de3-2530-4ee0-986e-cf0314334aec/p/868762F3-8E8F-4E23-B0DA-34C1783F0A03/canvas

**Comment by loripusey** [comment]
@wullaski where are we with this ticket?

**Comment by wullaski** [comment]
@loripusey I got this in PR now and working to get it merged in today. Just finishing up test coverage for it.
